File: app/src/main/java/in/androidmate/anujgupta/notes_room/ui/main/MainViewInterface.java
package in.androidmate.anujgupta.notes_room.ui.main;

import java.util.List;

import in.androidmate.anujgupta.notes_room.models.Note;

/**
 * Created by anujgupta on 14/01/18.
 */

public interface MainViewInterface {

    void onNotesLoaded(List<Note> notes);

    void onNoteAdded();

    void onDataNotAvailable();

}

File: app/src/main/java/dhbw/app_ent/demo/notes_room/ui/main/MainViewInterface.kt
package dhbw.app_ent.demo.notes_room.ui.main

import dhbw.app_ent.demo.notes_room.models.Note

interface MainViewInterface {
    open fun onNotesLoaded(notes: MutableList<Note?>?)
    open fun onNoteAdded()
    open fun onDataNotAvailable()
}